src/http_client/network_antenna/network_request.rs

```diff
@@ -19,6 +19,15 @@ impl NetworkRequest {
         }
     }
 
+    pub fn new_get(url: Url) -> Self {
```

Comment:
this type would benefit from a constructor, and change all other to delegate to that.

```rust
impl NetworkRequest {
    pub fn new(url: impl Into<Url>, method: impl Into<NetworkMethod>, headers: impl Into<HashMap<String, String>>, body: impl Into<BagOfBytes>) -> Self {
        Self {
           ....
        }
    }
}
```

And then:
`new_get` becomes:
```rust
pub fn new_get(url: Url) -> Self {
        Self::new(url, NetworkMethod::Get, [], [])
}
```
(also rewrite `new_post`!!)
🎉

Reply:
this sample seem to not work. I'm not sure how I would convert from [] to BagOfBytes